.github/workflows/semver.yml

```diff
@@ -0,0 +1,85 @@
+name: Semver Checks
+
+env:
+  CARGO_TERM_COLOR: always
+  CARGO_REGISTRIES_CRATES_IO_PROTOCOL: sparse
+
+on:
+  pull_request:
+    branches: [main]
+
+jobs:
+  changes:
+    name: Changed Crates
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: read
+    outputs:
+      crates: ${{ steps.filter.outputs.changes }}
+    steps:
+      - uses: dorny/paths-filter@v3
+        id: filter
+        with:
+          filters: |
+            ruma:
+              - 'crates/ruma/**'
+            ruma-appservice-api:
+              - 'crates/ruma-appservice-api/**'
+              - 'crates/ruma-common/**'
```

Comment:
Why check all the API crates when ruma-common changes?

Reply:
In case the `metadata!` macro changes.